### PR TITLE
7.x islandora 1851

### PR DESCRIPTION
**Pull request by nhart**
**JIRA Ticket**: https://jira.duraspace.org/browse/ISLANDORA-1851

# What does this Pull Request do?
This PR pages through the RI results vs returning them all at one.

# What's new?
Sorting by created date
Only querying object types that have a pattern set up.

# How should this be tested?
On a large data set, add a pattern for "All Islandora objects" and proceed to run the bulk updater. All Islandora objects should get an alias.

# Additional Notes:
* Does this change require documentation to be updated? No
* Does this change add any new dependencies?  No
* Does this change require any other modifications to be made to the repository (ie. Regeneration activity, etc.)? No
* Could this change impact execution of existing code? No

# Interested parties
@rosiel

### CODE REVIEW

File: islandora_pathauto.module [hunk 1]
```diff
@@ -143,49 +150,112 @@ function islandora_pathauto_islandora_object_purged($pid) {
...
-      $count = 0;
+    $context['finished'] = 0;
+    if ($sandbox['total'] == 0) {
+      $query_string = format_string($query, array(
+        '!filters' => implode(' ', $sandbox['filters']),
+        '!limit' => ''));
+      $sandbox['total'] = $tuque->repository->ri->countQuery($query_string, 'sparql');
+      if ($sandbox['total'] == 0) {
+        $context['finished'] = 1;
+        drupal_set_message(t('No objects found for selected modules (@models).', array('@models' => implode(', ', $sandbox['models']))));
```
Comment (rosiel) [inline islandora_pathauto.module]: Typo?  (modules)

**Review by rosiel — APPROVED** [review]
Ran the code. It works. I presume that doing it this way will prevent the issue in ISLANDORA-1851. Unless anyone thinks this warrants more thorough testing, I'm good to merge tomorrow.

File: islandora_pathauto.module (hunk 1 above)
Comment (nhart) [reply islandora_pathauto.module]: ha.. fixed.

**Comment by nhart** [comment]
I did run this a couple of times on approx. 28k objects. Worked fine

**Comment by nhart** [comment]
@Islandora/7-x-1-x-committers  any further thoughts on this?

**Comment by DiegoPino** [comment]
@nhart @rosiel saw you had this solved and was tested. Merging to squash some bugs out of the long list of open JIRA tickets. Please ping me if there are further comments or i did wrong.

**Comment by nhart** [comment]
thanks @DiegoPino

**Pull request merged**